### PR TITLE
fix(Hits): add test for template support

### PR DESCRIPTION
**Pull request by tkrugg**
This PR adds tests for make sure `ais-hits` works with `<ng-template>`
and that it's passing `hits` and `results` like we want to.

This is a fairly common use case and yet we do not test it.
I noticed this as I was attempting to write tests for the insights
integration.

### CODE REVIEW

File: src/hits/__tests__/hits.spec.ts [hunk 1]
```diff
@@ -4,38 +4,87 @@ import { NgAisHighlight } from '../../highlight/highlight';
...
+            <ng-template let-results="results">
+              {{ results | json }}
+            </ng-template>
+          </ais-hits>
+        `,
+      TestedWidget: NgAisHits,
+      additionalDeclarations: [NgAisHighlight],
+    });
+    const fixture = render({});
+    expect(fixture).toMatchSnapshot();
```
Comment (francoischalifour) [inline src/hits/__tests__/hits.spec.ts]: There's no way to test that the fixture's template was called with `results` instead of using a snapshot?

Comment (tkrugg) [reply src/hits/__tests__/hits.spec.ts]: I found a way but it's far too complicated.

```ts
    // 1. create mock component and extract instance in scope.
    let mockInstance;
    @Component({
      selector: 'mock-component',
      template: ``,
    })
    class MockComponent implements OnInit {
      @Input() hits: any;
      @Input() results: any;
      @Input() insights: any;
      ngOnInit() {
        mockInstance = this; 
      }
    }

    // 2. pass hits, results and other connector properties to template
    const render = createRenderer({
      defaultState,
      template: `
          <ais-hits>
            <ng-template let-hits="hits" let-results="results" let-insights="insights">
              <mock-component [hits]="hits" [results]="results" [insights]="insights"></mock-component>
            </ng-template>
          </ais-hits>
        `,
      TestedWidget: NgAisHits,
      additionalDeclarations: [NgAisHighlight, MockComponent],
    });

    const state = {
      ...defaultState,
      insights: jest.fn(),
    };

    // 3. trigger render
    render(state);

    // 4. assert
    expect(mockInstance.hits).toEqual(state.hits);
    expect(mockInstance.results).toEqual(state.results);
    expect(mockInstance.insights).toEqual(state.insights);
```

Another technique involved digging into what seems to be private properties of the fixture. 

For small json objects, snapshots are more straight forward.

Comment (francoischalifour) [reply src/hits/__tests__/hits.spec.ts]: That'd be nice if we could reuse this in a generic way!

Comment (tkrugg) [reply src/hits/__tests__/hits.spec.ts]: I'll keep looking into making this more reusable.
I guess we'll have to as we progress in allowing `ng-template` on more widgets and making it the first way to customize their UI.

**Pull request merged**